engine/src/flutter/lib/ui/semantics.dart

```diff
@@ -962,6 +1025,318 @@ class SemanticsFlag {
   String toString() => 'SemanticsFlag.$name';
 }
 
+/// Semantics flags.
+class SemanticsFlags {
+  /// All flags default to `false` unless specified.
+  const SemanticsFlags({
+    this.hasCheckedState = false,
+    this.isChecked = false,
+    this.isSelected = false,
+    this.isButton = false,
+    this.isTextField = false,
+    this.isFocused = false,
+    this.hasEnabledState = false,
+    this.isEnabled = false,
+    this.isInMutuallyExclusiveGroup = false,
+    this.isHeader = false,
+    this.isObscured = false,
+    this.scopesRoute = false,
+    this.namesRoute = false,
+    this.isHidden = false,
+    this.isImage = false,
+    this.isLiveRegion = false,
+    this.hasToggledState = false,
+    this.isToggled = false,
+    this.hasImplicitScrolling = false,
+    this.isMultiline = false,
+    this.isReadOnly = false,
+    this.isFocusable = false,
+    this.isLink = false,
+    this.isSlider = false,
+    this.isKeyboardKey = false,
+    this.isCheckStateMixed = false,
+    this.hasExpandedState = false,
+    this.isExpanded = false,
+    this.hasSelectedState = false,
+    this.hasRequiredState = false,
+    this.isRequired = false,
+  });
+
+  /// SemanticsFlags with everything default to false.
```

Comment:
nit: consider rephrasing, such as "The set of semantics flags with every flag set to false" (the word "default" implies a starting value that you can change later, but SemanticsFlags.none guaranteed that every flag is set to false and it cannot be changed)